roles/tarzan/tasks/main.yml

```
@@ -1,97 +1,99 @@
+    set_fact:
+      tarzan_http_port: "{{ tarzan_http_port_staging }}"
+  - name: set tarzan https port for staging
+    set_fact:
+      tarzan_https_port: "{{ tarzan_https_port_staging }}"
+  when: deployment_environment in [ "staging", "devel" ]
+
+- name: create tarzan root dir
+  file:
+    name: "{{ tarzan_root_dir }}"
+    state: directory
+    mode: g+rwXs
+
+- name: create tarzan conf dir
+  file:
+    name: "{{ tarzan_conf_dir }}"
+    state: directory
+    mode: g+rwXs
+
+- name: create tarzan log dir
+  file:
+    name: "{{ tarzan_log_dir }}"
+    state: directory
+    mode: g+rwXs
+
+- name: deploy server config file for kong
+  template:
+    src: "kong.conf.j2"
+    dest: "{{ tarzan_server_conf_file }}"
+
+- name: deploy entity config file for kong
+  template:
+    src: "kong.yml.j2"
+    dest: "{{ tarzan_entity_conf_file }}"
 
 - name: deploy ssl cert for kong
-  copy: src="tarzan_cert.pem" dest="{{ tarzan_conf }}/tarzan_cert.pem"
-
-- name: deploy ssl key for kong 
-  copy: src="tarzan_key.pem" dest="{{ tarzan_conf }}/tarzan_key.pem"
-
-# FIXME: Couldn't get this to work properly with supervisord because 
-# it demands that the program under control doesn't daemonize. 
-# (The kong binary itself spawns nginx and self binaries)
-- name: modify uppsala's crontab to start kong
-  lineinfile: dest="{{ ngi_pipeline_conf }}/crontab_upps"
-              line='# restart kong if it has died for some reason'
-              backup=no
-
-- name: modify uppsala's crontab to start kong
-  lineinfile: dest="{{ ngi_pipeline_conf }}/crontab_upps"
-              line='@reboot source $HOME/.bash_profile && kong start -c {{ tarzan_conf }}/webproxy.conf'
-              backup=no
-
-- name: modify uppsala's crontab to start kong
-  lineinfile: dest="{{ ngi_pipeline_conf }}/crontab_upps"
-              line='38 * * * *      source $HOME/.bash_profile && kill -0 $(cat {{ tarzan_log_dest }}/pids/nginx.pid) 2> /dev/null || kong start -c {{ tarzan_conf }}/webproxy.conf'
-              backup=no
-
-- name: snapshot cassandra once a week, so that normal backups can be taken of a sany copy of files (kb sizes)  
-  lineinfile: dest="{{ ngi_pipeline_conf }}/crontab_upps"
-              line='1 1 * * 1 {{ cassandra_dest }}/{{ cassandra_version }}/bin/nodetool -h localhost -p 7199 snapshot kong'
-              backup=no
-
-- name: dump cassandra schema once a week 
-  lineinfile: dest="{{ ngi_pipeline_conf }}/crontab_upps"
-              line='0 1 * * 1 export PATH=/usr/bin/:$PATH && {{ cassandra_dest }}/{{ cassandra_version }}/bin/cqlsh -e "DESC KEYSPACE kong" > {{ cassandra_db_dest }}/kong_schema-`date +%Y%m%d`.cql'
-              backup=no
+  copy:
+    src: "tarzan_cert.pem"
+    dest: "{{ tarzan_cert_file }}"
 
-- name: deploy script for easier stoping of kong services
-  template: src="stop_kong.sh.j2" dest="{{ ngi_resources }}/stop_kong.sh" mode=g+rwx
+- name: deploy ssl key for kong
+  copy:
+    src: "tarzan_key.pem"
+    dest: "{{ tarzan_key_file }}"
```

If you run ansible with `-D`, it will print out the file content if the task changed it. This might cause keys to end up in log files that may be accessible to a broader set of users than those who are allowed to read the actual key file.
Setting `no_log` on the task prevents that.
```suggestion
    dest: "{{ tarzan_key_file }}"
  no_log: True
```

---

Fixed